alacritty_terminal/src/clipboard.rs

```diff
@@ -18,35 +18,47 @@ use std::ffi::c_void;
 use log::{debug, warn};
 
 use copypasta::nop_clipboard::NopClipboardContext;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[cfg(all(not(any(target_os = "macos", windows)), feature = "wayland"))]
 use copypasta::wayland_clipboard;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[cfg(all(not(any(target_os = "macos", windows)), feature = "x11"))]
 use copypasta::x11_clipboard::{Primary as X11SelectionClipboard, X11ClipboardContext};
-use copypasta::{ClipboardContext, ClipboardProvider};
+#[cfg(any(feature = "x11", target_os = "macos", windows))]
+use copypasta::ClipboardContext;
+use copypasta::ClipboardProvider;
 
 pub struct Clipboard {
     clipboard: Box<dyn ClipboardProvider>,
     selection: Option<Box<dyn ClipboardProvider>>,
 }
 
 impl Clipboard {
-    #[cfg(any(target_os = "macos", target_os = "windows"))]
+    #[cfg(any(target_os = "macos", windows))]
     pub fn new() -> Self {
         Self::default()
     }
 
-    #[cfg(not(any(target_os = "macos", target_os = "windows")))]
-    pub fn new(display: Option<*mut c_void>) -> Self {
-        if let Some(display) = display {
-            let (selection, clipboard) =
-                unsafe { wayland_clipboard::create_clipboards_from_external(display) };
-            return Self { clipboard: Box::new(clipboard), selection: Some(Box::new(selection)) };
+    #[cfg(not(any(target_os = "macos", windows)))]
+    pub fn new(_display: Option<*mut c_void>) -> Self {
+        #[cfg(feature = "wayland")]
+        {
+            if let Some(display) = _display {
+                let (selection, clipboard) =
+                    unsafe { wayland_clipboard::create_clipboards_from_external(display) };
+                return Self {
+                    clipboard: Box::new(clipboard),
+                    selection: Some(Box::new(selection)),
+                };
+            }
         }
 
-        Self {
+        #[cfg(feature = "x11")]
+        return Self {
             clipboard: Box::new(ClipboardContext::new().unwrap()),
             selection: Some(Box::new(X11ClipboardContext::<X11SelectionClipboard>::new().unwrap())),
-        }
+        };
+
+        #[cfg(not(any(feature = "x11")))]
```

Comment:
Why is this `not(any(...))` of a single feature? Seems like it should be `#[cfg(not(any(feature = "x11", feature = "wayland")))]`

Reply:
Because that's the only reason why it could be reached, if `x11` is enabled, the `return Self { ... }` will always be hit and thus trigger a compiler warning about an unreachable statement.

So adding this fallback just for Wayland when X11 is disabled and no Display is present (so XWayland), should be fine.